0007 Canonical Metadata

dubadub opened the pull request:
[Rendered](https://github.com/cooklang/spec/blob/proposal/0007-canonical-metadata/proposals/0007-canonical-metadata.md)

This proposal introduces a standardised set of metadata keys for Cooklang, improving the way recipes are described and organised. By defining a core set of canonical metadata, we aim to ensure consistent, machine-readable annotations that simplify recipe organisation, sharing, and usability across tools while maintaining Cooklang's minimalist philosophy.

### Key Objectives:
- Provide clear, standardised metadata keys for common recipe attributes (e.g., preparation time, servings, author).
- Foster interoperability across tools by defining consistent and machine-readable metadata formats.
- Simplify recipe management and enhance the user experience for developers and consumers alike.

### Benefits:
- Unifies metadata across recipes, addressing common fragmentation issues such as inconsistent time formatting.
- Enhances automation, enabling tools to scale servings, sort by difficulty, and more.
- Offers a clear, optional framework for adding useful context to recipes without complicating Cooklang’s simplicity.

### Impact:
- **CookCLI**: Will include options to filter recipes based on these new metadata keys.
- **Mobile Apps**: Will incorporate search and display functionality for the new metadata, enhancing user interaction.

This proposal has been informed by ongoing discussions and feedback from the Cooklang community, and we invite your review and feedback on these enhancements.

### Relevant Discussions:
- [Add known/standard metadata keys to documentation](https://github.com/cooklang/spec/discussions/101)
- [Description, notes, and metadata...](https://github.com/cooklang/spec/discussions/46)
- [Use metadata for recipe picture URL](https://github.com/cooklang/spec/discussions/64)

jphastings commented:
This is excellent, thank you! Can I suggest that you establish a precedence between the metadata keys which have multiple options? For example, there should be a documented outcome if a recipe that contains the following two lines:

```cooklang
>> source: The Palomar Cookbook <urn:isbn:9781784720995>
>> author: Layo Paskin
```

I'd recommend something simple like the earliest conflicting declaration taking precedence.

Zheoni commented:
Already said some on discord, but here's all my thoughts/suggestions/questions:
- Allow to use the structure provided by YAML in the frontmatter
- I would keep source and author separate. Sometimes I want to link to a recipe and to the social media of the chef or whatever.
- `diet` could also take a list. A recipe may fit multiple diets.
- Lists of values like `tags` or `diet` (maybe) could also take a comma separated string.
- Is the `difficulty` value restricted to only `easy`, `medium` or `hard`?

dubadub commented:
I updated the proposal:
- added synonyms for nested YAML structures like `source.author`
- keep source and author separate.
- multiple values for `diet` and `image`
- added `introduction` and `description` for recipe intro.
- allow any values for `difficulty`, because for search purpose we can extract values user have
